Bottom: 0eafd583d6b966047f082798254d8f6cb2d1331d
Top:    35aac141c2713fbeb8df48e2953824e8b986df8e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-04 12:26:46 +0100

More hack to avoid GL calls when we're out of context


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 45b38eb..3ea1e05 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -475,6 +475,12 @@ ghid_set_color (hidGC gc, const char *name)
   double r, g, b, a;
   a = 1.0;
 
+  if (!check_gl_drawing_ok_hack)
+    {
+      current_color = NULL;
+      return;
+    }
+
   if (!alpha_changed && current_color != NULL)
     {
       if (strcmp (name, current_color) == 0)
